degree-importer/degree-importer.php

```diff
@@ -0,0 +1,595 @@
+     * @author Jim Barnes
+     * @since 1.0.0
+     * @param $name string | The program name
+     * @param $type string | The program type
+     * @param $graduate int | If the program is a graduate program
+     * @return string | The program suffix
+     **/
+    private function get_program_suffix( $name, $type, $graduate ) {
+        $lower_name = strtolower( $name );
+
+        switch( $type ) {
+            case 'minor':
+                return '-minor';
+            case 'certificate':
+                if ( stripos( $lower_name, 'certificate' ) === false ) {
+                    return '-certificate';
+                }
+            default:
+                return '';
+        }
+    }
+
+    /**
+     * Handles exceptions for college names
+     * @author Jim Barnes
+     * @since 1.0.0
+     * @param $college_name string | The college name from the search service
+     * @return string | The corrected college name
+     **/
+    private function get_college_name( $college_name ) {
+        $replacements = array(
+            'College of Hospitality Management' => 'Rosen College of Hospitality Management',
+            'Office of Undergraduate Studies' => 'College of Undergraduate Studies',
+            'College of Nondegree' => ''
+        );
+
+        if ( isset( $replacements[$college_name] ) ) {
+            return $replacements[$college_name];
+        }
+
+        return $college_name;
+    }
+
+    private function format_contacts( $contacts ) {
+        if ( ! $contacts ) {
+            return null;
+        }
+
+        $retval = array();
+
+        foreach( $contacts as $contact ) {
+            $split_name = explode( ',', $contact->contact_name );
+
+            $retval[] = array(
+                'degree_contact_name'  => $split_name[0],
+                'degree_contact_title' => trim( $split_name[1] ),
+                'degree_contact_phone' => $contact->contact_phone ? $contact->contact_phone : '',
+                'degree_contact_email' => $contact->contact_email ? $contact->contact_email : ''
+            );
+        }
+
+        return $retval;
+    }
+
+    /**
+     * Handles matching undergraduate degrees
+     * to the undergraduate catalog data
+     * @author Jim Barnes
+     * @since 1.0.0
+     * @param $program Object | The program object
+     * @return (string|NULL) | The catalog url if available.
+     **/
+    private function get_catalog_url( $program ) {
+        $clean_program_name = $this->clean_name( $program->name );
+        $clean_college_name = $this->clean_name( $program->college_name );
+
+        if ( $this->catalog_programs ) {
+            foreach( $this->catalog_programs as $key => $uc_program ) {
+                $uc_clean_program_name = $this->clean_name( $uc_program->name );
+                $uc_clean_type_name = $this->clean_name( $uc_program->type );
+                $uc_clean_college_name = $this->clean_name( $uc_program->college );
+
+                if (
```

Comment:
Comments would be helpful to explain what's going on here:

https://github.com/UCF/Main-Site-Theme/blob/master/jobs/degree-importer.php#L150-L157